src/ophys_etl/modules/mesoscope_splitting/__main__.py

```diff
@@ -263,6 +324,64 @@ def run(self):
         duration = time.time()-t0
         self.logger.info(f"that took {duration:.2e} seconds")
 
+    def create_stitched_full_field_image(self):
```

Comment:
Add doc string.

I get what you are doing here, that is checking for everything needed to create a "stitched_full_field_image". However, the logic here of returning None to block off the real code reads as messy to me. My suggestion is to split this into two internal functions: The first performs and returns true/false if one the conditions are met for the full field code (or not) second would actually create the full field on true. So two functions, one `check` one `create`. I think it would also make your tests a bit simpler perhaps.

Reply:
Because the checks I'm doing happen while I am constructing `full_field_path` from the platform json, I implemented a method that returns `full_field_path` or `None`.

Because the `create` method you describe was going to be a very thin wrapper around `write_out_stitched_full_field_image`, I just moved that logic into `run()`.